### PR TITLE
🌱 Verify e2e code compiles as part of pull-cluster-api-provider-aws-build

### LINKED ISSUE #1429: CI presubmits should make sure that test/e2e/... compiles
#1426 passed all presubmits but broke compilation of test/e2e. We should verify that all the test code can build and compile in our presubmits (but leave running the tests to other jobs).

### PR DESCRIPTION
**Pull request by varunpurohit76**
**What this PR does / why we need it**:
added compile-e2e

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1429

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -160,7 +160,7 @@ test-conformance-fast: ## Run clusterctl based conformance test on workload clus
 ## Binaries
 ## --------------------------------------
 .PHONY: binaries
-binaries: managers clusterawsadm ## Builds and installs all binaries
+binaries: managers clusterawsadm compile-e2e ## Builds and installs all binaries
```
Comment (ncdc) [inline Makefile:163]: I think it might be better to remove this and instead update scripts/ci-build.sh to have it `make binaries compile-e2e`. WDYT?

Comment (varunpurohit76) [reply Makefile:163]: could be done that way as well if we want to keep compilation of test separate.
Question is, should `.PHONY: binaries` make sure that all essential items compile or we want to keep that control outside?

Comment (ncdc) [reply Makefile:163]: I would consider `binaries` to cover the executables that are user-facing and not include testing.

Comment (varunpurohit76) [reply Makefile:163]: makes sense. will push the update.

**Comment by ncdc** [comment]
LGTM. Could you please squash your commits?

**Comment by varunpurohit76** [comment]
@ncdc all squashed into a single commit.

**Comment by varunpurohit76** [comment]
@ncdc @randomvariable let me know if any input is required from my side..

**Pull request merged**